### PR TITLE
47946 Optional writing of activities in the agents log with plugin forms & fields.

### PR DESCRIPTION
**Pull request by yeneastgate**
related to https://github.com/onOffice-Web-Org/oo-wp-plugin/issues/809
changed log:
Add activity option & create agents log when create forms

### CODE REVIEW

**Comment by fredericalpers** [comment]
@yeneastgate please resolve the conflicts here, thank you :)

**Comment by fredericalpers** [comment]
@yeneastgate When activating the feature I get the following error and all settings I made are deleted: 

An error occurred while saving the form. Please check whether the form name is unique.

The form definitely only exist once. Please investitage and fix the error, thank you :)

**Comment by yeneastgate** [comment]
> @yeneastgate When activating the feature I get the following error and all settings I made are deleted:
> 
> An error occurred while saving the form. Please check whether the form name is unique.
> 
> The form definitely only exist once. Please investitage and fix the error, thank you :)

@fredericalpers Could you please try downgrading the plugin version to 48 to check again? The issue seems to occur only when the wp_oo_plugin_form_activityconfig table has not been added.

**Comment by fredericalpers** [comment]
@yeneastgate Thank you for implementing this feature, it's working well so far! 👍

However, there are a couple of adjustments and additions that are still needed:

- Checkbox functionality: As long as the checkbox to enable the feature is not checked, all related fields should be disabled to prevent unintended input.
- Additional selection fields: Please integrate the fields Origin Contact (origincontact) and Advisory Level (advisorylevel) as selection options so that these can also be reflected in the activity.

Once these changes are made, the feature should be fully ready for use.

Thank you in advance for the further adjustments!

**Comment by fredericalpers** [comment]
Vorgezogen auf 5.2

**Comment by yeneastgate** [comment]
> * Checkbox functionality: As long as the checkbox to enable the feature is not checked, all related fields should be disabled to prevent unintended input.

@fredericalpers After checking, I cannot reproduce the error as you described. You might be experiencing browser cache issues. Could you please try clearing your browser cache?

**Comment by yeneastgate** [comment]
> * Additional selection fields: Please integrate the fields Origin Contact (origincontact) and Advisory Level (advisorylevel) as selection options so that these can also be reflected in the activity.

@fredericalpers I would like to confirm with you some points: 
1. For the option "advisory level", we will implement in 2 forms: "Contact form in the detail view" and "Owner form" because only these 2 forms have "estateId".

2. Do you agree with the placement of the fields "Origin Contact (origincontact)" and "Advisory Level (advisorylevel)" in the "Activities" box as shown in the attached image?

image

**Comment by fredericalpers** [comment]
@yeneastgate please go ahead and implement the suggested changes, thank you :)

**Comment by yeneastgate** [comment]
@fredericalpers Currently, I have checked that when the API is called with `advisorylevel` set to G, the `reasoncancellation` field is used to specify the reason for cancellation. However, when calling the API to get fields, we are unable to retrieve the values for the `reasoncancellation` field.

In the video, I tested that `reasoncancellation` only accepts newly created values. 

https://github.com/user-attachments/assets/755e0138-3a59-4fa4-9182-67c6c594d55d

Do we need to include `reasoncancellation` when selecting option G? If so, could you provide the API details for it?

**Comment by fredericalpers** [comment]
@yeneastgate the <code>reasoncancellation</code> does not have to be implemented for now. I will send a ticket to the IT department. Thank you :)

**Comment by yeneastgate** [comment]
> Edit: It's actually not possible for this to be retrieved?

@fredericalpers  Yes, currently I can't retrieve when using the API get field.


> reasoncancellation STRING. OPTIONAL. Reason of cancellation (Absagegrund). Functions together with advisory level G = Cancellation. Values can be found in enterprise under “Extras >> Settings >> Administration >> Singleselect >> Module: Property Activities” >> key field: Reason of cancellation” in the column “content”.
> 
> https://apidoc.onoffice.de/actions/datensatz-anlegen/maklerbuch-aktivitaeten/

I've read the document but can't retrieve the "Reason of cancellation" field, as shown in the attached image.
image

**Comment by yeneastgate** [comment]
> * Checkbox functionality: As long as the checkbox to enable the feature is not checked, all related fields should be disabled to prevent unintended input.
> * Additional selection fields: Please integrate the fields Origin Contact (origincontact) and Advisory Level (advisorylevel) as selection options so that these can also be reflected in the activity.

@fredericalpers I have integrated the Origin Contact (origincontact) and Advisory Level (advisorylevel) fields. Please take a look at my demo video. Thanks!

https://github.com/user-attachments/assets/e617ca48-6135-481d-bd05-ec49f59ab773

**Comment by fredericalpers** [comment]
just fyi: There is currently no API call with which you can read out the possible values for rejection reasons. the hardcoded solution must suffice.

File: css/admin.css [hunk 1]
```diff
@@ -528,6 +528,11 @@ body #post-head-content p.wp-clearfix:nth-child(1) {
 	color: #999;
 }
 
+.select2-dropdown.select2-dropdown--below,
+.select2-dropdown.select2-dropdown--above {
+	margin-top: 0 !important;
+}
```
Comment (andernath) [inline css/admin.css:534]: Is this global change necessary? I don't see the difference. Could you please tell me? :)

Comment (yeneastgate) [reply css/admin.css]: @andernath There has been a change in the process, now it is no longer necessary. thank you

**Pull request merged**